lib/realtime/tenants/authorization/policies/broadcast_policies.ex

```diff
@@ -24,74 +23,64 @@ defmodule Realtime.Tenants.Authorization.Policies.BroadcastPolicies do
           write: boolean()
         }
   @impl true
-  def check_read_policies(_conn, policies, %Authorization{channel: nil}) do
+  def check_read_policies(_conn, policies, %Authorization{channel_name: nil}) do
     {:ok, Policies.update_policies(policies, :broadcast, :read, false)}
   end
 
-  def check_read_policies(conn, %Policies{} = policies, %Authorization{
-        channel: %Channel{id: channel_id}
-      }) do
-    Postgrex.transaction(conn, fn transaction_conn ->
-      query = from(b in Broadcast, where: b.channel_id == ^channel_id)
+  def check_read_policies(conn, %Policies{} = policies, %Authorization{channel_name: channel_name}) do
+    query =
+      from(m in Message,
+        where: m.channel_name == ^channel_name,
+        where: m.feature == :broadcast,
+        limit: 1
```

Comment:
is the channel_name not unique?

Reply:
no, we will have multiple messages from the same channel stored in this table